testing/travis.py

```diff
@@ -125,10 +124,17 @@ def run_cmds(cmds, fail_sequential=False):
     return to_return
 
 
-def filter_not_docfiles(diff_names):
+def skip(diff_names):
+    if os.environ['TRAVIS_BRANCH'] != 'master' and os.environ["TRAVIS_PULL_REQUEST"]:
+        logging.info("We don't want to build on pushes to branches that aren't master.")
+        return True
 
     is_doc = lambda f: f.startswith('docs') or any([f.endswith(ext) for ext in ['md', 'rst']])
-    return [f for f in diff_names.split() if not is_doc(f)]
+    if len([ f for f in diff_names.split() if not is_doc(f) ]) < 1:
+        logging.info("All changed files were for documentation.")
+        return True
+
+    return False
```

Comment:
Two more things we can check:
 1. Are the changes entirely in addons/ ? If so, don't build
 2. Are the changes to `.gitignore`, `.mention-bot`, or another file that isn't documentation but we don't want to build on? If so, skip the build.